tests/test_instantiate_config.py

```diff
@@ -0,0 +1,78 @@
+import unittest
+from datetime import date
+
+import hydra
+from omegaconf import DictConfig
+
+from core_lib.client.client_base import ClientBase
+from core_lib.core_lib import CoreLib
+from core_lib.data_layers.data.handler.sql_alchemy_data_handler_registry import SqlAlchemyDataHandlerRegistry
+from core_lib.error_handling.status_code_exception import StatusCodeException
+from core_lib.helpers.config_instances import instantiate_config
+from examples.test_core_lib.core_lib.data_layers.data.db.user import User
+
+dattime = date.today()
+user_data = {
+    'email': 'jon@email.com',
+    'username': 'jondoe',
+    'password': 'jon@12345',
+    'nick_name': 'jon',
+    'first_name': 'Jon',
+    'middle_name': 'P',
+    'last_name': 'Doe',
+    'birthday': dattime,
+    'gender': User.Gender.MALE.value,
+}
+
+
+class ExampleCoreLib(CoreLib):
+    def __init__(self, conf: DictConfig):
+        super().__init__()
+        self.config = conf
+        config_file = 'instantiate_sql_alchemy.yaml'

```

Comment:
this is something I have never done before., not sure to recommend users this way of solving problems.

1.  if you want to add config add them to the original config passed to this core-lib. 
    We can use the set_default provided by hydra. 
2. the original config already contains the config for the SQL alchemy connection. why another
3. make sure the examples in the tests are making sense.